### DISCUSSION TITLE
Manual annotation of ROIS in CaImAn?

### DISCUSSION THREAD

**Discussion by AyusoAa** [flatironinstitute/CaImAn#1154] [General]
Hi CaImAn users & developers! In my experience using CNMF-E for 1p data, ROI recognition is never perfect and often some neurons are missed, especially when there is high heterogeneity in shapes and/or dynamics. It would be convenient if CaImAn would allow for manually adding those missing ROIs ( basically drawing them) to subsequently perform CNMF-E again. Is this a feature that other users would appreciate having incorporated into the software? on the developer side, would this be feasible?

**Comment by j-friedrich** [comment]
That feature has already been incorporated. To use semi-automatic initialization you have to set the `'seed_method'` to `'semi'` and make sure that `init_iter` is greater than 1 (CaImAn will perform `init_iter-1` automatic initialization rounds and the last one is manual). Note that this does not support parallel processing, i.e. use `dview=None` when creating the `CNMF` object and don't forget to use a backend that will pop up a regular plot window, e.g. with `%matplotlib qt`.

**Reply by EricThomson** [reply]
This would be a good demo notebook I'll add it to the list I think it would be really helpful to demonstrate, maybe even with a video to show. Maybe at the workshop I'll touch base with you Johannes so I can work on it a bit with you to make sure I've got the details right, and then I'll work on a video for it.

**Comment by AyusoAa** [comment]
Dear both, thank you so much for the tips. For sure a demo and/or video would be useful for those of us still struggling to get fully comfortable with CaImAn and CNMF-E. We will give it a try incorporating those in the parameters dictionary within our .py script which we have written adapting demo_pipeline_cnmfE.ipynb to our computational resources.  And we will follow issue #1130 closely since we would also like to import masks form Fiji.
